### PR TITLE
Task10

### PR DESCRIPTION
**Pull request by sachidanandaEpam**
Pending item:
- Not able to link store to session storage
- I am not able load currently edited item from store

### CODE REVIEW

**Comment by sachidanandaEpam** [comment]
image

File: src/app/core/guards/auth.guard.ts [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
-import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { CanActivate, UrlTree } from '@angular/router';
+import { Store, select } from '@ngrx/store';
 import { Observable } from 'rxjs';
-import { tap } from 'rxjs/operators';
-import { AuthService } from '../services/auth.service';
+import { map, take } from 'rxjs/operators';
+import { AuthActions } from '../store/actions';
+import { AuthStates } from '../store/state';
+import * as AuthReducer from '../store/reducers';
```
Comment (MariMax) [inline src/app/core/guards/auth.guard.ts]: you could import only `selectLoggedIn`  like `import {selectLoggedIn} from '../store/reducers'` and i'd probably made selectors in a separate folder or file

File: src/app/core/services/auth.service.ts [hunk 2]
```diff
@@ -1,44 +1,46 @@
...
+      }),
+      switchMap(response => this.getInitialAuthState())
+    );
   }
 
-  public logout(): void {
-    this._session.removeAccessToken();
-    this._isAuthenticated = false;
-    this._router.navigate(['/login']);
+  public logout(): Observable<boolean> {
```
Comment (MariMax) [inline src/app/core/services/auth.service.ts]: not sure why do you want Observable from it, but there is an easier way
```
this._session.removeAccessToken();
return of(true);
```

File: src/app/core/store/reducers/auth.reducer.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { AuthActions } from '../actions';
+import { AuthStates } from '../state';
+
+export const reducer = createReducer(
+  AuthStates.initialAuthState,
+  on(AuthActions.loginSuccess, (state, { authState }) => ({ ...state, isAuthenticated: authState.isAuthenticated, user: authState.user })),
+  on(AuthActions.logout, () => AuthStates.initialAuthState)
+);
+
+export const getUser = (state: AuthStates.IAuthState) => state.user;
```
Comment (MariMax) [inline src/app/core/store/reducers/auth.reducer.ts]: I'd move selectors into separate file

File: src/app/core/store/reducers/items.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+export const reducer = createReducer(
+  ItemStates.initialItemState,
+  on(ItemActions.loadItemsSuccess, (state: ItemStates.IItemState, { items, criteria }) => ({ ...state, items, criteria })),
+  on(ItemActions.deleteItemSuccess, (state: ItemStates.IItemState, { deletedItem }) => ({ ...state, deletedItem })),
+  on(ItemActions.loadItemByIdSuccess, (state: ItemStates.IItemState, { selectedItem }) => ({ ...state, selectedItem })),
+  on(ItemActions.createItemSuccess, (state: ItemStates.IItemState, { createdItem }) => ({ ...state, createdItem })),
+);
+
+export const getItems = (state: ItemStates.IItemState) => state.items;
```
Comment (MariMax) [inline src/app/core/store/reducers/items.reducer.ts]: also please move selectors into separate file

File: src/app/core/services/auth.service.ts (hunk 2 above)
Comment (sachidanandaEpam) [reply src/app/core/services/auth.service.ts]: I wanted to handle the failure scenario if logout fails. I have planned to handle it in future and improve upon it.

**Pull request merged**